src/key.rs

```diff
@@ -450,6 +451,12 @@ impl PublicKey {
             PublicKey(pk)
         }
     }
+    /// Creates a new public key from a [`ElligatorSwift`] and the global [`SECP256K1`] context.
```

Comment:
```suggestion
    /// Creates a new public key from an [`ElligatorSwift`].
```

`ElligatorSwift::ellswift_decode` uses `ffi::secp256k1_context_no_precomp` .

Reply:
That's actually a good observation. **Should** it use `secp256k1_context_no_precomp`? I think it make sense because the ES encode/decode only uses **field** operations, and AFAIK, the precomp helps with **group** operations. But I might be wrong on that, any thought @apoelstra?

Reply:
Yep, if no_precomp works (and it definitely will if this is really just field ops, no ecmults) then we should use it.